browser/ui/views/toolbar/brave_side_panel_toolbar_button.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/views/toolbar/brave_side_panel_toolbar_button.h"
+
+#include <memory>
+
+#include "base/memory/raw_ptr.h"
+#include "brave/components/constants/pref_names.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/views/toolbar/side_panel_toolbar_button.h"
+#include "chrome/browser/ui/views/toolbar/toolbar_button.h"
+#include "components/prefs/pref_service.h"
+#include "ui/base/models/simple_menu_model.h"
+#include "ui/views/context_menu_controller.h"
+
+namespace {
+class SidePanelMenuModel : public ui::SimpleMenuModel,
+                           public ui::SimpleMenuModel::Delegate {
+ public:
+  explicit SidePanelMenuModel(PrefService* prefs)
+      : ui::SimpleMenuModel(this), prefs_(prefs) {
+    Build();
+  }
+
+  ~SidePanelMenuModel() override = default;
+  SidePanelMenuModel(const SidePanelMenuModel&) = delete;
+  SidePanelMenuModel& operator=(const SidePanelMenuModel&) = delete;
+
+ private:
+  enum ContextMenuCommand { HideSideBarButton };
+
+  void ExecuteCommand(int command_id, int event_flags) override {
+    if (command_id == ContextMenuCommand::HideSideBarButton)
+      prefs_->SetBoolean(kShowSidePanelButton, false);
+  }
+
+  void Build() {
```

Comment:
optional: Also loose convention, a class's own method should be declared first and then overridden method should come.

```
void Build() { }
...

// SimpleMenuModel::Delegate:
void Execute...
```

Reply:
Cheers!